src/Databases/DatabaseFileSystem.cpp

```diff
@@ -0,0 +1,141 @@
+#include <Databases/DatabaseFileSystem.h>
+
+#include <IO/Operators.h>
+#include <IO/WriteBufferFromString.h>
+#include <Interpreters/Context.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ParserCreateQuery.h>
+#include <Parsers/parseQuery.h>
+#include <Poco/File.h>
+#include <Poco/Path.h>
+#include <Storages/IStorage.h>
+#include <TableFunctions/TableFunctionFactory.h>
+
+
+namespace DB
+{
+
+DatabaseFileSystem::DatabaseFileSystem(const String & name_, const String & path_, ContextPtr context_)
+    : IDatabase(name_), WithContext(context_->getGlobalContext()), path(path_), log(&Poco::Logger::get("DatabaseFileSystem(" + name_ + ")"))
+{
+    if (path.empty())
+        path = Poco::Path::current();
+}
+
+std::string DatabaseFileSystem::getTablePath(const std::string& table_name) const
+{
+    return Poco::Path(path, table_name).toString();
+}
+
+void DatabaseFileSystem::addTable(const std::string& table_name, StoragePtr table_storage) const
+{
+    std::lock_guard lock(mutex);
+    loaded_tables.emplace(table_name, table_storage);
+}
+
+bool DatabaseFileSystem::isTableExist(const String & name, ContextPtr) const
+{
+    {
+        std::lock_guard lock(mutex);
+        if (loaded_tables.find(name) != loaded_tables.end())
+            return true;
+    }
+
+    Poco::File table_file(getTablePath(name));
```

Comment:
please use `std::filesystem` instead of `Poco::File`, we aim to do that everywhere. Please also fix this in all other places.

Reply:
fixed